README.md

```diff
@@ -1,4 +1,102 @@
+A small Python library and CLI that aids in FHIR model development.
+
+Some features are:
+
+1. **Validation** - the CLI wraps the Java-based [HL7 FHIR IG Publisher](https://confluence.hl7.org/display/FHIR/IG+Publisher+Documentation) and provides a more user friendly interface to
+   it. Also adds additional validation to ensure consistency in FHIR resource
+   file names and payloads.
+2. **Publish the model** - POST a directory of FHIR resources to a FHIR server
+3. **JSON/XML conversion** - Convert FHIR resource payloads between JSON and XML
+
+## Installation
+1. Git clone this repository
+
+```bash
+$ git clone git@github.com:ncpi-fhir/ncpi-fhir-utility.git
+$ cd ncpi-fhir-utility
+```
+
+2. Setup a Python virtual environment
+
+```bash
+# Create virtualenv
+$ python3 -m venv venv
+
+# Activate virtualenv
+$ source ./venv/bin/activate
+```
+
+3. Install the necessary requirements
+
+```bash
+$ pip install -e .
+```
+Test the installation by running the CLI: `fhirutil -h`. You should see
+something that contains:
+```
+Usage: fhirutil [OPTIONS] COMMAND [ARGS]...
+
+  A CLI utility for validating FHIR Profiles and Resources
+```
+
+4. Install Docker CE: https://docs.docker.com/install/
+
+Docker is needed because the `fhirutil` CLI executes the model validation
+inside a Docker container which runs the HL7 FHIR IG Publisher.
+
+## Model Validation
+
+The `fhirutil` validation feature is meant to be used on a folder which follows
+the required structure and content of a FHIR ImplementationGuide. The
+phrase "FHIR model" refers to the conformance resources in the
+ImplementationGuide.
+
+### Setup
+
+```shell
+# Copy a FHIR resource into the test IG
+cp tests/data/profiles/valid/StructureDefinition-participant.json tests/data/site_root/input/resources/profiles
+```
+
+### Validate
+You can use the CLI in two different ways to aid in validating your model:
+
+1. Use the CLI to do all steps of validation:
+
+- Add necessary configuration for the FHIR resources in your model to the
+  ImplementationGuide FHIR resource
+- Validate file names and FHIR resource IDs
+- Invoke the Dockerized version of the IG Publisher to validate the model
+
+```shell
+$ fhirmodel validate tests/data/site_root --publisher_opts='-tx n/a'
+```
+
+2. Use the CLI along with the native IG Publisher:
+
+- Add necessary configuration for the FHIR resources in your model to the
+  ImplementationGuide FHIR resource
+- Validate file names and FHIR resource IDs
+- Run the IG Publisher Java jar natively on your machine to validate the model
+
+```shell
+$ fhirutil add ./tests/data/site_root/input/resources
+$ java -jar org.hl7.fhir.publisher.jar -ig site_root/ig.ini -tx n/a
+```
+
+This method is a bit faster since it's running the IG Publisher on your machine
+and does not need to spin up a Docker container.
+
+See [instructions](https://confluence.hl7.org/display/FHIR/IG+Publisher+Documentation#IGPublisherDocumentation-Installing) for installing
+the IG publisher.
```

Comment:
```suggestion
This method is a bit faster since it's running the IG Publisher directly and does not need to spin up a Docker container, but it means that you must first install the IG Publisher yourself ([IG Publisher installation instructions](https://confluence.hl7.org/display/FHIR/IG+Publisher+Documentation#IGPublisherDocumentation-Installing)).
```